src/cr/cube/dimension.py

```diff
@@ -156,287 +159,481 @@ def is_mr_selections(self, others):
+        # Either get rid of the need for that alternate return value type or
+        # create a separate method for it.
+        elements = (
+            self._all_elements if include_missing else self._valid_elements
+        )
+
+        include_subtotals = (
+            include_transforms and
+            self.dimension_type != 'categorical_array'
+        )
+
+        interleaved_items = tuple(self._iter_interleaved_items(elements))
+
+        labels = list(
+            item.label
+            for item in interleaved_items
+            if include_subtotals or not item.is_insertion
+        )
+
+        if include_cat_ids:
+            element_ids = tuple(
+                None if item.is_insertion else item.element_id
+                for item in interleaved_items
+                if include_subtotals or not item.is_insertion
             )
-            for label in labels_with_cat_ids
-            if self._include_in_labels(label, valid_indices)
-        ]
+            return list(zip(labels, element_ids))
+
+        return labels
 
     @lazyproperty
     def name(self):
         """Name of a cube's dimension."""
-        refs = self._dim['references']
+        refs = self._dimension_dict['references']
         return refs.get('name', refs.get('alias'))
 
+    @lazyproperty
+    def numeric_values(self):
+        """tuple of numeric values for valid elements of this dimension.
+
+        Each category of a categorical variable can be assigned a *numeric
+        value*. For example, one might assign `like=1, dislike=-1,
+        neutral=0`. These numeric mappings allow quantitative operations
+        (such as mean) to be applied to what now forms a *scale* (in this
+        example, a scale of preference).
+
+        The numeric values appear in the same order as the
+        categories/elements of this dimension. Each element is represented by
+        a value, but an element with no numeric value appears as `np.nan` in
+        the returned list.
+        """
+        return tuple(
+            element.numeric_value for element in self._valid_elements
+        )
+
     @lazyproperty
     def shape(self):
-        return len(self._elements)
+        return len(self._all_elements)
 
     @lazyproperty
-    def subtotals(self):
-        view = self._dim.get('references', {}).get('view', {})
+    def _all_elements(self):
+        """_AllElements object providing cats or subvars of this dimension."""
+        return _AllElements(self._dimension_dict['type'])
+
+    def _iter_interleaved_items(self, elements):
+        """Generate element or subtotal items in interleaved order.
+
+        This ordering corresponds to how value "rows" (or columns) are to
+        appear after subtotals have been inserted at their anchor locations.
+        Where more than one subtotal is anchored to the same location, they
+        appear in their document order in the cube response. pairs for this
+        dimension.
```

Comment:
This last sentence fragment is an editing error. I've fixed it in my copy and will push with any other changes before we proceed.